### PR TITLE
fixing sqw_eval for spherical projection

### PR DESCRIPTION
**Pull request by abuts**
Addresses Re #1533 -- second part.
bunch of minor bugs fixing `sqw_eval` (and `bose`) for spherical projection.

Nothing more to tell -- unit tests confirming bug  presense now passes so it is the fixture.

Summary of changes (summary by CM from PR review)
- the Bose factor method has been corrected, and a test added that now passes
- in bin_pixels, the shape of the accumulated npix1 array is properly controlled
- handling of the ub_inf_legacy method is altered.
- uniformity of the binary_op_manager methods' interface has been improved
- efficiency of the binary_op managers has been improved by allowing pixel sorting not to be performed when not needed
- comments have been improved

### CODE REVIEW

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]
A summary of the changes has  been added to the PR description. Please review. It is still not so clear what fixes point 2 of the issue, what is a bug encountered while dealing with this, and what is just improved efficiency and documentation.

File: horace_core/sqw/coord_transform/@aProjectionBase/private/transform_pix_to_hkl_.m [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 end
 
-if ~isempty(obj.ub_inv_legacy) % legacy alignment
+if ismethod(obj,'ub_inf_legaccy') && ~isempty(obj.ub_inv_legacy) % legacy alignment
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/coord_transform/@aProjectionBase/private/transform_pix_to_hkl_.m:30]: good point!

File: horace_core/sqw/axes_blocks/@AxesBlockBase/private/bin_pixels_.m [hunk 2]
```diff
@@ -153,13 +153,20 @@
             pix_indx(on_edge(:,i),i) = n_bins(i);
         end
     end
+    reconcile_sizes = false;
     if numel(n_bins) == 1
         n_bins = [n_bins,1];
+        reconcile_sizes = true;
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/axes_blocks/@AxesBlockBase/private/bin_pixels_.m:159]: hm, do not understand this suggestion. code below needs nbins as `size` and `size`  always 2D array, not [1,1]. The added boolean needed for reconciling row/column transition as it is apparently matters here. (code fails)

File: herbert_core/utilities/classes/@data_op_interface/binary_op_manager_single.m [hunk 3]
```diff
@@ -22,7 +22,20 @@
 %   <#file:> <doc_file_sigvar_notes>
 % <#doc_end:>
 % -----------------------------------------------------------------------------
+% Optional:
+% ignore_cell_sorting
+%             By default false and binary operation which includes pixels
+%             sort pixels with every bin when operation is performed.
```
[the remark this answers is not in the document]
Comment (abuts) [reply herbert_core/utilities/classes/@data_op_interface/binary_op_manager_single.m]: this is fundamental feature of Horace, last explained on the first slide of the discussion we have about `page_op`

Pixels in Horace are arranged (sorted) according to bins but within the bin they may be ordered randomly. 
I do not think the comment on particular operation should (or could) describe principal structure of `sqw` object.

[the remark this answers is not in the document]
Comment (abuts) [reply herbert_core/utilities/classes/@data_op_interface/binary_op_manager_single.m]: actually, it is not difficult to write this again -- just two centences.

File: _test/test_sqw_pageOpMethods/test_eval.m [hunk 4]
```diff
@@ -20,7 +20,18 @@
...
+        end
+        %
+        function test_func_eval_with_shpere_proj(obj)
+            spp = sphere_proj;
+            spp.offset = [0,1.5,0];
+            sp_cut = cut(obj.sqw_obj,spp,[0,0.01,2],[0,180],[-180,180],0);
+
+            % Almost zero bose correction in this case. Just to check
+            % algorighm works
+            spBose = bose(sp_cut,8);
```
[the remark this answers is not in the document]
Comment (abuts) [reply _test/test_sqw_pageOpMethods/test_eval.m:32]: The test verifies that Bose works now. (was failing in the past)
Testing Bose correctness is not the purpose of this ticket as it needs scientific validation. And proper data. Here I took random piece of data and see that Bose works but 0 for this piece. I think the scope of this ticket is already big enough to add scientific validation to it.

File: herbert_core/utilities/classes/@data_op_interface/data_op_interface.m [hunk 5]
```diff
@@ -57,7 +57,8 @@
         wout = mtimes  (w1,w2)
         wout = plus    (w1,w2)
         %------------------------------------------------------------------
-        w = binary_op_manager_single(w1, w2, op_function_handle);
+        w = binary_op_manager(w1,w2,op_function_handle,varargin);
+        w = binary_op_manager_single(w1, w2, op_function_handle,varargin);
```
[the remark this answers is not in the document]
Comment (abuts) [reply herbert_core/utilities/classes/@data_op_interface/data_op_interface.m:61]: I do normally place cemicolumns to declarations. I believe, old code may not have them in some places. 

If you think this is important, we should discuss it, make agreement and follow this agreement.

**Review by cmarooney-stfc — APPROVED** [review]

**Pull request merged**